integration/app/test-demo/package.json

```diff
@@ -0,0 +1,37 @@
+{
+  "name": "test-demo",
+  "version": "0.1.0",
+  "dependencies": {
```

Comment:
Should these dependencies be separated into devDependencies and dependencies or match our library package.json or demo one?

I think it might look like this:
```
"devDependencies": {
    "@types/react": "^17.0.33",
    "@types/react-dom": "^17.0.10",
    "@types/react-router-dom": "^5.3.2",
    "@types/styled-components": "^5.1.15",
    "@types/webpack-env": "^1.16.3",
    "@typescript-eslint/eslint-plugin": "^5.2.0",
    "@typescript-eslint/parser": "^5.2.0",
    "aws-sdk": "^2.1015.0",
    "concurrently": "^6.3.0",
    "html-webpack-plugin": "^5.5.0",
    "react-dev-utils": "^11.0.4",
    "react-router-dom": "^5.3.0",
    "source-map-loader": "^3.0.0",
    "ts-loader": "^9.2.6",
    "typescript": "^4.4.4",
    "uuid": "^8.3.2",
    "webpack-cli": "^4.9.1",
    "webpack-dev-server": "^4.3.1"
},
"dependencies": {
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "styled-components": "^5.0.0",
    "styled-system": "^5.1.5"
    "amazon-chime-sdk-component-library-react": "^2.11.1",
    "amazon-chime-sdk-js": "^2.20.0",
}
```

May be follow the existing demo: https://github.com/aws-samples/amazon-chime-sdk/blob/main/apps/meeting/package.json

Reply:
Do we have any convention for this? I'm not good at how to split these two things.
I assume those dependencies directly relate to SDK itself should be "dependencies", those for webpack, eslint, script, environment are for "devDependencies".
Those "@types/*" should be in "dependencies" I guess.
Could you please share your knowledge here? : )

Reply:
I made some investigation online. In general, this is a `package.json` for a demo App where from my opinion doesn't has a very clear gap between prod environment and dev environment. So it doesn't matter very much. (Correct me if I'm wrong)

I updated the `package.json` as you suggested. It makes sense to me.
```
  "dependencies": {
    "amazon-chime-sdk-component-library-react": "^2.11.1",
    "amazon-chime-sdk-js": "^2.20.0",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "styled-components": "^5.3.3",
    "styled-system": "^5.1.5"
  },
  "devDependencies": {
    "@types/react": "^17.0.33",
    "@types/react-dom": "^17.0.10",
    "@types/react-router-dom": "^5.3.2",
    "@types/styled-components": "^5.1.15",
    "@types/webpack-env": "^1.16.3",
    "aws-sdk": "^2.1015.0",
    "concurrently": "^6.3.0",
    "html-webpack-plugin": "^5.5.0",
    "react-dev-utils": "^11.0.4",
    "react-router-dom": "^5.3.0",
    "source-map-loader": "^3.0.0",
    "ts-loader": "^9.2.6",
    "typescript": "^4.4.4",
    "uuid": "^8.3.2",
    "webpack-cli": "^4.9.1",
    "webpack-dev-server": "^4.3.1"
  },
```

Reply:
Sorry for the delayed response.
This is a general good practice to differentiate devDependencies and dependencies:
1. Major difference is builders of this test app know what is required to develop the app vs what is required to run the app.
2. There is also difference when it comes to installation. One point I know of is the transitive dependency installation. For dependencies, all package version should be resolved and satisfy each others version to work if interdependent. This is not the case for devDependencies I think.

Hence IMO we should not differentiate between a test demo and a library but follow general best practices.